misk/src/main/kotlin/misk/clustering/lease/LeaseManager.kt

```diff
@@ -0,0 +1,14 @@
+package misk.clustering.lease
+
+interface LeaseManager {
+  /**
+   * Registers interest in the lease with the given name. Service instances should register their
+   * interest in leases as soon as they know about them; this gives the process an opportunity
+   * to notify the underlying cluster system that it is interested in the lease, or to setup
+   * other background tasks to acquire the lease if necessary. Note that registering interest
+   * in a lease doesn't necessarily mean that this service instance will attempt to acquire
+   * the lease; many clustering systems will only try to acquire leases for resources that
+   * they think they should own based on some consistent hashing system
+   */
+  fun requestLease(name: String): Lease
```

Comment:
I’m not sure on this API. As a lease user I’m expecting methods like `acquire()` and `release()`, this API has a weaker `acquire()` and no `release()`.

Should we call this something else?

Reply:
That's intentional. You don't explicitly `acquire` and `release` a lease - you register interest in it, and if the clustering system decides you are the instance that it should be assigned to, then in the background it is acquired. We're modeling load balanced leases here, where all processes in the cluster indicate that they are potential candidates for owning a lease, and the actual distribution of named lease ownership is handled by the clustering system.

Reply:
That said, I'm open to other names.

Reply:
I merged with the current name, but I'm also not happy with it so if you have other suggestions, I'm down to change it.

Reply:
Thumbs up. I don’t have better names either.